atc/exec/set_pipeline_step.go

```diff
@@ -0,0 +1,229 @@
+package exec
+
+import (
+	"context"
+	"fmt"
+	"github.com/concourse/baggageclaim"
+	"io/ioutil"
+	"os"
+	"sigs.k8s.io/yaml"
+	"strings"
+
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/lager/lagerctx"
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/atc/creds"
+	"github.com/concourse/concourse/atc/db"
+	"github.com/concourse/concourse/atc/db/lock"
+	"github.com/concourse/concourse/atc/exec/artifact"
+	"github.com/concourse/concourse/vars"
+)
+
+//go:generate counterfeiter . SetPipelineDelegate
+
+type SetPipelineDelegate interface {
+	BuildStepDelegate
+
+	Initializing(lager.Logger)
+	Starting(lager.Logger)
+	Finished(lager.Logger, ExitStatus, db.ConfigVersion, db.ConfigVersion)
+}
+
+// TaskStep executes a TaskConfig, whose inputs will be fetched from the
+// artifact.Repository and outputs will be added to the artifact.Repository.
+type SetPipelineStep struct {
+	planID      atc.PlanID
+	plan        atc.SetPipelinePlan
+	metadata    StepMetadata
+	delegate    SetPipelineDelegate
+	dbConn      db.Conn
```

Comment:
In general, we've tried to stay away from exposing the db directly to each step type. Instead we use the delegate as a proxy to any database operations.

For example: 
https://github.com/concourse/concourse/blob/f6ec622e57d6559fffd676a5df16665745600a7a/atc/exec/check_step.go#L161